src/pyPolarionCli/__main__.py

```diff
@@ -108,40 +113,45 @@ def add_parser() -> argparse.ArgumentParser:
                         help="Print full command details before executing the command.\
                             Enables logs of type INFO and WARNING.")
 
-    # to do: Register subparsers once the command files are generated.
-
     return parser
 
 
-def main() -> int:
+def main() -> Ret:
     """ The program entry point function.
 
     Returns:
         int: System exit status.
     """
     ret_status = Ret.OK
+    commands = []
 
-    # Parse the command line arguments.
+    # Create the main parser and add the subparsers.
     parser = add_parser()
+    subparser = parser.add_subparsers(required='True', dest="cmd")
+
+    # Register all commands.
+    for cmd_register in _COMMAND_REG_LIST:
+        cmd_par_dict = cmd_register(subparser)
+        commands.append(cmd_par_dict)
+
+    # Parse the command line arguments.
     args = parser.parse_args()
 
+    # Check if the command line arguments are valid.
     if args is None:
         ret_status = Ret.ERROR_ARGPARSE
+        parser.print_help()
     else:
         # If the verbose flag is set, change the default logging level.
         if args.verbose:
             logging.basicConfig(level=logging.INFO)
-
-        logging.info("Program arguments: ")
-
-        for arg in vars(args):
-            logging.info("* %s = %s", arg, vars(args)[arg])
+            logging.info("Program arguments: ")
+            for arg in vars(args):
```

Comment:
Not a serious problem, but:
change is not necessary, since the logging level is set to logging.INFO
therefore, the logging.info("...") will only be shown if verbose flag is set

Reply:
It will not be shown but still, it belongs to the information under the verbose section